### ISSUE TITLE
[figma] Migrate from "Tokens Studio for Figma" to Figma variables

### ISSUE THREAD

**Issue by oliviertassinari** [mui/mui-design-kits#198]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

**Comment by dusskapark** [comment]
Hi @oliviertassinari 

I tried the conversion by applying the example of converting json-2-variables added to Figma's official plugin sample though, I found that there were a lot of errors unfortunately. 

When I checked it, Token Studio officially supports Figma Variable conversion only for the paid version, and announced that there are the following limitations so far. I think it would be good for you to check the contents.

- https://docs.tokens.studio/variables/creating-variables

**Comment by adrianmanea** [comment]
We are doing the conversion manually. Everything works well so far.

**Comment by oliviertassinari** [comment]
The limitations to Figma variables they mention in https://docs.tokens.studio/variables/creating-variables are interesting.

**Comment by adrianmanea** [comment]
While I agree with the limitations they mentioned, I have not encountered any obstacles that would prevent us from delivering a good experience for the designers. Dark/Light mode works 100x better natively.

The only downsides I can see for now is that global changes for typography still require manual changes.

**Comment by oliviertassinari** [comment]
> The only downsides I can see for now is that global changes for typography still require manual changes.

So maybe we need to use both? Figma variables and Tokens Studio for Figma?

**Comment by adrianmanea** [comment]
> So maybe we need to use both? Figma variables and Tokens Studio for Figma?

I've been playing with this idea to see if there are conflicts. If we can achieve having both would cover more ground. For most users the native Variables should suffice since the UX is significantly better. I'm hoping David will have some free time to help us with setting up a docs app dedicated for the library? 🤔 Related issue: https://github.com/mui/mui-design-kits/issues/138

**Comment by oliviertassinari** [comment]
> I'm hoping @DavidCnoops will have some free time to help us with setting up a docs app dedicated for the library? thinking Related issue: #138

@adrianmanea You have write access to https://www.notion.so/mui-org/MUI-for-Figma-docs-4eb22360388047a9a31fc19d300a285e is there something specific you would like to see better documented

**Comment by adrianmanea** [comment]
> You have write access
Great news! I need to document why we cannot support both Tokens Studio and Native Variables. Components can only be bound by either a classic style or a variable, and Tokens Studio only works with styles.

**Comment by dusskapark** [comment]
@adrianmanea 
Let me explain the solution I came up with. It is possible to connect Variable to the existing classic style. 

Well... having both variables and the classic styles in the list can make it messy, so it is not recommended. But I hope to provide tips for previous version customers like me who want to create a backward-compatible Figma env. :) 

<img width="601" alt="Screenshot 2023-07-04 at 8 11 34 PM" src="https://github.com/mui/mui-design-kits/assets/4177529/15778149-e219-4d19-b800-ff424fc65b56">

**Comment by adrianmanea** [comment]
Oh, wow! That is pretty powerful. I did not know that. Unfortunately, you cannot change the opacity to a variable as a style. One workaround could be to add a white or black layer on top of the variable to simulate opacity, but is that correct? Having a duplicate style of a variant still feels unnecessary. 

Additionally, Tokens Studio overwrites the whole set once you apply it the first time, which can be confusing.

**Comment by dusskapark** [comment]
@adrianmanea 

Yeah, I also had a similar token studio overwriting problem. fxxk

Figma's Variables are not yet providing variables for typography or opacity values, which is a big problem. 
However, as a consumer, I do not want to leave a dependency on the existing token studio to solve those issues. Because realistically, we need to subscribe to the paid plan of Token Studio to use both the Token Studio and the new Variable...

I heard that you are creating a separate plugin that automatically generates theme.js files. I would like to keep the tokens.json file independent of the existing token studio and include it in MUI operated plugin. 


You may already know this sample code, but I'm forwarding the link as a reference.

https://github.com/figma/plugin-samples/blob/master/variables-import-export

**Comment by adrianmanea** [comment]
I don't think supporting both options is even an option since components are bound to the variables now, not styles; this means that even if you'd want to use the Tokens Studio one would need to remap all components from using variables back to styles.

Can Tokens be bound to variables now? It is getting super confusing.

Typography variables are coming next: https://help.figma.com/hc/en-us/articles/4406787442711

**Comment by olivier-g74** [comment]
Bonjour Olivier, dear Ardian,
I was wondering why Colorstyles/_library clickableLayer, colorHighlight and fill Highlight are still in Local styles in 6.1.0 and not migrated to Local variables/Palettes.
I believe it was a simple omission or is there a deeper reason behind it ?
Thank you !
Olivier![Image](https://github.com/user-attachments/assets/fe6babde-6e0f-456f-9dc8-16bc93acfdc0)
